packages/background/src/controllers/NetworkController.ts

```diff
@@ -781,6 +831,101 @@ export default class NetworkController extends BaseController<NetworkControllerS
         }
     }
 
+    /**
+     *  switchProvider
+     *
+     *  @param chainId chain identifier of the network
+     *  @param providerType type of provider to use (default, backup or custom)
+     *  @param customRpcUrl rpc url of the network
+     *
+     */
+    public async switchProvider(
+        chainId: number,
+        providerType: ProviderType,
+        customRpcUrl?: string
+    ): Promise<void> {
+        const network = this.getNetworkFromChainId(chainId);
+        if (!network) throw new Error('Network not found for given chainId');
+
+        let rpcUrl;
+        switch (providerType) {
+            case ProviderType.DEFAULT:
+                rpcUrl = network.defaultRpcUrl;
+                break;
+            case ProviderType.BACKUP: {
+                rpcUrl = await this._getWorkingBackupRpcUrl(chainId);
+                break;
+            }
+            case ProviderType.CUSTOM:
+                rpcUrl = customRpcUrl;
+                break;
+            default:
+                throw new Error('Invalid provider type');
+        }
+        if (!rpcUrl) throw new Error('Invalid rpc url');
+
+        await this.editNetwork(chainId, {
+            rpcUrls: [rpcUrl],
+            blockExplorerUrls: network.blockExplorerUrls,
+            name: network.desc,
+            test: network.test,
+        });
+
+        this.store.updateState({
+            providerStatus: {
+                ...this.getState().providerStatus,
+                isUsingBackupProvider: providerType === ProviderType.BACKUP,
+            },
+        });
+    }
+
+    /**
+     *  _isRpcValid
+     *
+     *  Checks if the rpc url is working
+     *  @param chainId chain identifier of the network
+     *  @param rpcUrl rpc url of the network
+     *  @returns true if the rpc url is valid, false otherwise
+     */
+    private async _isRpcValid(
+        chainId: number,
+        rpcUrl: string
+    ): Promise<boolean> {
+        let isValid = false;
+        try {
+            const returnedChainId = await getCustomRpcChainId(rpcUrl);
+            if (returnedChainId === chainId) {
```

Comment:
for the sake of simplicity, I think is better to put:

```
...
            const returnedChainId = await getCustomRpcChainId(rpcUrl);
            isValid = returnedChainId === chainId
...
```

Reply:
Agreed! will go with this 

```
        try {
            const returnedChainId = await getCustomRpcChainId(rpcUrl);
            return returnedChainId === chainId;
        } catch (error) {
            return false;
        }
```